Decisions.TruBot/Steps/BotSteps.cs

```
@@ -0,0 +1,268 @@
+                Sid = ModuleSettingsAccessor<TruBotSettings>.GetSettings().Sid
+            };
+
+            try
+            {
+                JobIdRequest inputs = new JobIdRequest();
+                inputs.JobId = jobId;
+                
+                JsonContent content = JsonContent.Create(inputs);
+                
+                string result = TruBotRest.TruBotPost($"{baseUrl}/GetJobStatus", auth, content);
+                
+                return JobStatusResponse.JsonDeserialize(result);
+            }
+            catch (Exception ex)
+            {
+                throw new BusinessRuleException("The request to TruBot was unsuccessful.", ex);
+            }
+        }
+        
+        public JobStatusResponse GetJobStatusByJobExecutionId(string jobExecutionId,
+            [IgnoreMappingDefault, PropertyClassification(0, "Override Base URL", "Settings")] string? overrideBaseUrl)
+        {
+            if (string.IsNullOrEmpty(jobExecutionId))
+            {
+                throw new BusinessRuleException("jobExecutionId cannot be null or empty.");
+            }
+
+            string baseUrl = ModuleSettingsAccessor<TruBotSettings>.GetSettings().GetBotUrl(overrideBaseUrl);
+            
+            TruBotAuthentication auth = new TruBotAuthentication
+            {
+                Token = ModuleSettingsAccessor<TruBotSettings>.GetSettings().Token,
+                Sid = ModuleSettingsAccessor<TruBotSettings>.GetSettings().Sid
+            };
+
+            try
+            {
+                JobExecutionIdRequest inputs = new JobExecutionIdRequest();
+                inputs.JobExecutionId = jobExecutionId;
+
+                JsonContent content = JsonContent.Create(inputs);
+                
+                string result = TruBotRest.TruBotPost($"{baseUrl}/GetJobStatus", auth, content);
+                
+                return JobStatusResponse.JsonDeserialize(result);
```

I'm trying to imagine how these errors will appear.
If there's an error in the Post, that does not return a result,
the exception will be based on trying to serialize a null object, rather than the Post response, right?

Otherwise, this TruBotPost method, should bubble an exception here.

---

If something goes wrong in the Post message, it will return a business rule exception with a related status code, like 401 (Unauthorized) for instance